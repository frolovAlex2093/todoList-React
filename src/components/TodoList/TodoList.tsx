import React from 'react';

import { Box } from '@mui/material';

import type { Todo } from '../../pages/TodoHome';
import { Panel } from '../Panel/Panel';

import { TodoItem } from './TodoItem/TodoItem';

interface TodoListProps {
  editTodoId: Todo['id'] | null;
  todoList: Todo[];
  onChangeTodo: ({ name, description }: Omit<Todo, 'id' | 'checked'>) => void;
}

export const TodoList: React.FC<TodoListProps> = ({ todoList, editTodoId, onChangeTodo }) => (
    <Box>
      {todoList.map((todo) => {
        if (todo.id === editTodoId)
          return <Panel mode='edit' onChangeTodo={onChangeTodo} editTodo={todo} key={todo.id} />;
        return <TodoItem key={todo.id} todo={todo} />;
      })}
    </Box>
  );
