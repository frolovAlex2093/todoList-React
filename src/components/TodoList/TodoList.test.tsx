import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { TodoList } from "./TodoList";

const todoList = [
  {
    id: 1,
    name: "test name 1",
    description: "test description 1",
    checked: false,
  },
  {
    id: 2,
    name: "test name 2",
    description: "test description 2",
    checked: false,
  },
  {
    id: 3,
    name: "test name 3",
    description: "test description 3",
    checked: false,
  },
];

const editTodoId = null;

const onDeleteTodo = jest.fn();
const onCheckTodo = jest.fn();
const onEdit = jest.fn();
const onClickMore = jest.fn();
const onChangeTodo = jest.fn();

describe("TodoList component", () => {
  it("TodoList render", () => {
    render(
      <MemoryRouter>
        <TodoList
          todoList={todoList}
          
          
          
          
          editTodoId={editTodoId}
          onChangeTodo={onChangeTodo}
        />
      </MemoryRouter>
    );
    expect(screen.getByText("test name 1")).toBeInTheDocument();
    expect(screen.getByText("test description 1")).toBeInTheDocument();
    expect(screen.getByText("test name 2")).toBeInTheDocument();
    expect(screen.getByText("test description 2")).toBeInTheDocument();
    expect(screen.getByText("test name 3")).toBeInTheDocument();
    expect(screen.getByText("test description 3")).toBeInTheDocument();
  });

  it("TodoList snapshot", ()=>{
    const todoListSanp = render(
        <MemoryRouter>
          <TodoList
            todoList={todoList}
            editTodoId={editTodoId}
            onChangeTodo={onChangeTodo}
          />
        </MemoryRouter>
      );

      expect(todoListSanp).toMatchSnapshot()
  })
});
