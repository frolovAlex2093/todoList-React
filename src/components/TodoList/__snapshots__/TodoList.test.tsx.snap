// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component TodoList snapshot 1`] = `
Array [
  Object {
    "checked": false,
    "description": "test description 1",
    "id": 1,
    "name": "test name 1",
  },
  Object {
    "checked": false,
    "description": "test description 2",
    "id": 2,
    "name": "test name 2",
  },
  Object {
    "checked": false,
    "description": "test description 3",
    "id": 3,
    "name": "test name 3",
  },
]
`;
